Add example about how to dump heap

yejianfengblue opened the pull request:
## Proposed changes

For Appium user who is unfamilar with Node.js (like me), they usually start appium server by running the executable `bin/appium` and  don't know how to add the command line parameter `--heapsnapshot-signal=<signal>` to `node` process which executes the appium.js script.

## Types of changes

What types of changes does your code introduce to Appium?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/appium/appium/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla.js.foundation/appium/appium)
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

yejianfengblue committed:
Add example about how to dump heap

For Appium user who is unfamilar with Node.js (like me), they usually start appium server by running the executable `bin/appium` and  don't know how to add the command line parameter `--heapsnapshot-signal=<signal>` to `node` process which executes the appium.js script.

mykola-mokhnach commented on docs/en/advanced-concepts/memory-collection.md:
````diff
@@ -21,6 +21,21 @@ kill -SIGUSR2 &lt;nodePID&gt;
+#### Examples
+```shell
+# Go to the directory where appium is installed via npm. 
+cd /home/k/.npm-global/lib/node_modules/appium/
````
`cd "$(npm -g root)/lib/appium/"` or `cd "$(npm root)/lib/appium/"`depending on which npm root is needed (global or local)

yejianfengblue replied:
How about `cd "$(npm -g root)/appium/"` or `cd "$(npm root)/appium"`?

```shell
# In my Manjaro linux where npm prefix is set
[k@a2600 ~]$ npm --version
6.14.7
[k@a2600 ~]$ npm config list | grep prefix
prefix = "/home/k/.npm-global"
[k@a2600 ~]$ echo "$(npm -g root)"
/home/k/.npm-global/lib/node_modules
[k@a2600 ~]$ ls "$(npm -g root)"
appium  appium-base-driver  appium-doctor  face  mjpeg-consumer  node  node-gyp  opencv4nodejs


# In my Manjaro linux where npm prefix is not set
[k@x15 ~]$ npm --version
6.14.7
[k@x15 ~]$ echo "$(npm -g root)"
/usr/lib/node_modules
```

mykola-mokhnach replied:
^ what is the issue with the commands above? Everything looks correct

yejianfengblue replied:
I mean use `cd "$(npm -g root)/appium/"` instead of `cd "$(npm -g root)/lib/appium/"`, because `$(npm -g root)` already contains the path `/lib/node_modules`. And directory `appium` is directly under `/lib/node_modules`.

mykola-mokhnach replied:
ah, ok, this was actually a copypaste typo from my side

mykola-mokhnach commented on docs/en/advanced-concepts/memory-collection.md:
```diff
@@ -21,6 +21,21 @@ kill -SIGUSR2 &lt;nodePID&gt;
+# Go to the directory where appium is installed via npm. 
+cd /home/k/.npm-global/lib/node_modules/appium/
+# start appium server
+node --heapsnapshot-signal=SIGUSR2 . # don't miss the dot in the end
```
process PID could also be saved right here: `pid=$!`

KazuCocoa reviewed (approved):
seems good after fixing the comment

yejianfengblue committed:
Use smarter commands in dump heap example

As suggested by @mykola-mokhnach,
1. use `cd "$(npm -g root)/appium/"` or `cd "$(npm root)/appium/"` rather than hardcode my example appium directory path.
2. get the node process PID using `pid=$!` rather than `ps a | grep node`, so that the variable `pid` can be referenced later.
